src/main/java/uk/gov/digital/ho/hocs/notify/application/RequestData.java

```diff
@@ -16,17 +17,33 @@
     static final String USER_ID_HEADER = "X-Auth-UserId";
     static final String GROUP_HEADER = "X-Auth-Groups";
 
-    private static final String ANONYMOUS = "anonymous";
-
     private static boolean isNullOrEmpty(String value) {
         return value == null || value.equals("");
     }
 
+    public void parseMessageHeaders(Map<String,String> headers) {
```

Comment:
we can replace this with `headers.forEach(MDC::put);` but I feel this is more explicit